esp32_ulp/__main__.py

```diff
@@ -10,7 +10,7 @@
 
 def src_to_binary(src):
     assembler = Assembler()
-    assembler.assemble(src)
+    assembler.assemble(src, remove_comments=False)  # comments already removed by preprocessor
     garbage_collect('before symbols export')
     addrs_syms = assembler.symbols.export()
     for addr, sym in addrs_syms:
```

Comment:
a typical use of this code is to import src_to_binary from here and call it with some source as str argument.

the main function is not used in this case, so the preprocess step should be moved from there to here.

Reply:
the comment then also makes more sense for this use case. :-)

Reply:
Changed.